frame/support/src/storage/types/counted_map.rs

```diff
@@ -134,7 +134,10 @@ where
 
 	/// Store or remove the value to be associated with `key` so that `get` returns the `query`.
 	pub fn set<KeyArg: EncodeLike<Key>>(key: KeyArg, q: QueryKind::Query) {
-		<Self as MapWrapper>::Map::set(key, q)
+		match QueryKind::from_query_to_optional_value(q) {
```

Comment:
We could add a migration which scans a `CountedStorageMap` and corrects the counter.  
Would probably need to run something manual against the current chains to check if we do that.

Reply:
> We could add a migration which scans a `CountedStorageMap` and corrects the counter.

@ggwpez should we just create a ticket to add a migration, gets this merged and have a follow-up?


> Would probably need to run something manual against the current chains to check if we do that.

Not sure what this means, could you clarify?

Reply:
>should we just create a ticket to add a migration, gets this merged and have a follow-up?

Yes sure.

>>Would probably need to run something manual against the current chains to check if we do that.
>Not sure what this means, could you clarify?
AFAIK we dont have any reflection to iterate through all `CountedStorageMap`s in the runtime. It is also not exposed in the runtime metadata… so not sure how we would programmatically loop through all maps.